src/components/app/data/hooks/usePassLearnerCsodParams.test.jsx

```diff
@@ -0,0 +1,97 @@
+import { renderHook } from '@testing-library/react-hooks';
+import { QueryClientProvider } from '@tanstack/react-query';
+import { useLocation, useParams } from 'react-router-dom';
+import { enterpriseCustomerFactory } from '../services/data/__factories__';
+import useEnterpriseCustomer from './useEnterpriseCustomer';
+import { queryClient } from '../../../../utils/tests';
+import { updateUserCsodParams } from '../services';
+import usePassLearnerCsodParams from './usePassLearnerCsodParams';
+
+jest.mock('./useEnterpriseCustomer');
+jest.mock('../services', () => ({
+  ...jest.requireActual('../services'),
+  updateUserCsodParams: jest.fn().mockResolvedValue(null),
+}));
+jest.mock('react-router-dom', () => ({
+  ...jest.requireActual('react-router-dom'),
+  useParams: jest.fn(),
+  useLocation: jest.fn(),
+}));
+const mockEnterpriseCustomer = enterpriseCustomerFactory();
+const mockedUserGuid = 'test-guid';
+const mockedSessionToken = 'test-token';
+const mockedCallbackUrl = 'mock-url';
+const mockedSubdomain = 'test-subdomain';
+const mockedSearchString = `?userGuid=${mockedUserGuid}&sessionToken=${mockedSessionToken}&callbackUrl=${mockedCallbackUrl}&subdomain=${mockedSubdomain}`;
```

Comment:
nit: would it be helpful / more readable to generate this string with `new URLSearchParams` and `queryParams.toString()`?